src/Symfony/Component/Messenger/Transport/KernelTerminate/KernelTerminateTransport.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\KernelTerminate;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+class KernelTerminateTransport implements TransportInterface
+{
+    private $handlers = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        $this->handlers[] = $handler;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stop(): void
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope)
+    {
+    }
+
+    public function __invoke()
+    {
+        foreach ($this->handlers as $message) {
```

Comment:
Instead, it should call the `receive` method and provide the `$handler`. Similar to what is done in [`Worker`](https://github.com/symfony/symfony/blob/b399527ba1c5c20f8b0f4db2b6563304c31c38ac/src/Symfony/Component/Messenger/Worker.php#L42-L48). But that mean the transport instance should somehow be aware of the bus where to dispatch the message to be handled (should the bus name be part of the DSN/transport options and injected by the factory?).